### PR TITLE
replaced MSTest with NUnit

### PR DESCRIPTION
**Pull request by enikao**
* Supports parameterized tests on class level (required for multiple counterparts)
* Timeout works (almost always)
* SetUp / TearDown work
* Nice inheritance of attributes (less duplication)
* Executes a bit faster

### CODE REVIEW

**Comment by enikao** [comment]
@JojoEffect Could you have a look at this NUnit setup?

File: src/cs/LionWeb.Integration.WebSocket.Tests/LionWeb.Integration.WebSocket.Tests.csproj [hunk 1]
```diff
@@ -5,10 +5,9 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Microsoft.NET.Test.Sdk"/>
+        <PackageReference Include="Microsoft.NET.Test.Sdk" />
```
Comment (dslmeinte) [inline src/cs/LionWeb.Integration.WebSocket.Tests/LionWeb.Integration.WebSocket.Tests.csproj:8]: (superfluous and inconsistent extra space 😄 )

File: src/cs/LionWeb.Integration.WebSocket.Tests/SetupTrace.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// // SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+
+namespace LionWeb.Integration.WebSocket.Tests;
+
+[SetUpFixture]
+[Timeout(WebSocketTestBase.TestTimeout)]
+[NonParallelizable]
+public class SetupTrace
```
Comment (dslmeinte) [inline src/cs/LionWeb.Integration.WebSocket.Tests/SetupTrace.cs:25]: For this kind of boilerplate code, it'd be really useful to document its purpose – probably a framework-related configuration – in ~1 sentence.

Also: it should be either `SetUpTrace` or `TraceSetup` 😺

File: src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs [hunk 3]
```diff
@@ -18,39 +18,66 @@
...
+    OtherCSharp
+}
 
-    protected WebSocketClientTestBase(LionWebVersions? lionWebVersion = null, List<Language>? languages = null) : base(lionWebVersion, languages)
-    {
-        Debug.WriteLine(Directory.GetCurrentDirectory());
-        StartServer();
-    }
+[TestFixture(ServerProcesses.CSharp)]
+// [TestFixture(ServerProcesses.OtherCSharp)]
```
[the remark this answers is not in the document]
Comment (dslmeinte) [reply src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs:35]: Would be good to have this comment as documentation here.

File: src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs [hunk 4]
```diff
@@ -74,29 +101,35 @@ private void StartServer()
...
 
-    /// <inheritdoc />
-    [TestCleanup()]
-    [ClassCleanup(InheritanceBehavior.BeforeEachDerivedClass, ClassCleanupBehavior.EndOfClass)]
-    public void Dispose()
+    private static readonly List<Process> _allServers = []; 
+    
+    [TearDown]
+    [OneTimeTearDown]
+    public void StopServer()
```
Comment (dslmeinte) [inline src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs:114]: `StopServers` (plural)?

File: src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketTestBase.cs [hunk 5]
```diff
@@ -25,7 +24,7 @@ namespace LionWeb.Integration.WebSocket.Tests;
 
 public abstract class WebSocketTestBase
 {
-    private const int TestTimeout = 6000;
+    public const int TestTimeout = 6000;
```
Comment (dslmeinte) [inline src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketTestBase.cs:27]: I'd almost say this could go in the `SetupTrace` file.)

**Review by dslmeinte — APPROVED** [review]
(but see questions)

File: src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (enikao) [reply src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs:35]: I see this as temporary state -- very soon we'll have another server to enter for real.

File: src/cs/LionWeb.Integration.WebSocket.Tests/SetupTrace.cs (hunk 2 above)
Comment (enikao) [reply src/cs/LionWeb.Integration.WebSocket.Tests/SetupTrace.cs:25]: Most of this code, including the name, is "inspired" 1:1 from NUnit docs

Comment (dslmeinte) [reply src/cs/LionWeb.Integration.WebSocket.Tests/SetupTrace.cs:25]: Can you add a link to those docs, or provide a 1 sentence summary?

File: src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs (hunk 4 above)
Comment (enikao) [reply src/cs/LionWeb.Integration.WebSocket.Tests/WebSocketClientTestBase.cs:114]: In principle we only want to stop one server (as these are highlander-servers). The edge case of several servers only happens if we execute several tests together, and for some reason the previous one has not been killed.
